cli/src/cli.rs

```diff
@@ -0,0 +1,97 @@
+use clap::{Parser, Subcommand};
+
+/**
+Welcome to the Simperby CLI!
+*/
+#[derive(Debug, Parser)]
+#[clap(name = "git")]
+#[clap(about = "A Simperby client CLI", long_about = None)]
+pub struct Cli {
+    #[clap(subcommand)]
+    command: Commands,
+}
+
+#[derive(Debug, Subcommand)]
+pub enum Commands {
+    /// Initialize a new Simperby node in the current directory.
+    Init,
+    /// Sync the `main` branch to the given commit.
+    ///
+    /// This will verify every commit along the way.
+    /// If the given commit is not a descendant of the
+    /// current `main` (i.e., cannot be fast-forwarded), it fails.
+    ///
+    /// Note that if you sync to a block `H`, then the `main` branch will move to `H-1`.
+    /// To sync the last block `H`, you have to run `update`.
+    /// (This is because that the finalization proof for a block appears in the next block.)
+    Sync { commit: String },
+    /// Print the information about the block data Git server that this node is hosting.
+    Git,
+    /// Clean the repository, removing all the outdated (incompatible with `main`) branches.
+    Clean {
+        /// If enabled, it will remove all the branches except `main`.
+        #[clap(long, action)]
+        hard: bool,
+    },
+    /// Vote on the agenda, broadcasting to the network.
+    /// It will also leave a `vote` tag on the commit (with some postfix).
+    Vote { commit: String },
+    /// Propose the given agenda or block.
+    ///
+    /// 1. If the commit is the last transaction,
+    /// it will create an 'agenda' commit and append on top,
+    /// immediately broadcast it to the network, and automatically vote on it.
+    ///
+    /// 2. If the commit is the last extra-agenda transaction,
+    /// it will create a `chat` and 'block' commit and append on top,
+    /// leaving a `proposal` tag on the commit.
+    /// This will be broadcasted in running the command `consensus`
+    /// if this node becomes the consensus leader then.
+    ///
+    /// 3. If the commit is a block, the `proposal` tag will be moved to the commit.
+    Propose { commit: String },
+    /// Veto the block, leaving a `veto` tag on the commit (with some postfix).
+    ///
+    /// It will be broadcasted to the network in the next `update`, if the conditions are met.
+    /// It fails if the given commit is already set to `proposal`.
+    /// If the given commit is already set to `veto`, it will be removed.
+    Veto { commit: String },
```

Comment:
Is this the nil-vote to veto the lazy leader?

Reply:
Yes.

Reply:
I think I missed the case of veto to the round, not a specific block. I added.